### PR TITLE
API/UI for collaborator roles

### LINKED ISSUE #80: Add roles to collaborators
https://instedd.storiesonboard.com/m/chatbots/!card/RYj6dwTYC4mTWjnh8KeDNA

- [x] add roles to collaborators model: Publish, Behaviour, Content, Variables, Results.
  - Note: Publish < Behaviour. Behaviour < Content. Behaviour < Variables. Publish < Results.
- [x] add roles to invitations when they are created/sent
- [x] show roles in collaborators
- [x] allow edition of roles of collaborators (todo: check if a dialog is the right way)
- [x] disabled (but do not hide) areas where the user can't access.
  - [x] corner case, some that can access just variables, needs to access the variables page through translations.
- [x] Channel page can't be shown for users with only behaviour/content/variables/results roles. Publish is needed to edit the channel.
- any collaborator can preview/test the bot
  - [x] corner case: if user can only edit content, the behaviour is forbidden, but the preview button needs to be there.
- [x] the only role that can add/invite collaborators is the admin
- [x] download manifest requires publish, otherwise any role should be able to read the behavior & channel configuration.
- [x] duplicate bot requires being the owner

- leave the logic of permissions inheritance in the server side. send boolean for the required permissions checks when loading the bot for the current user.

- default bot page might depend on user permissions

### PR DESCRIPTION
**Pull request by ggiraldez**
Closes #80 

Also adds resending invitations from the UI

### CODE REVIEW

File: app/models/collaborator.rb [hunk 1]
```diff
@@ -25,4 +25,10 @@ def user_is_not_bot_owner
       errors[:user_id] << "is the owner of the bot"
     end
   end
+
+  def roles_are_valid
+    unless roles.blank? || roles.all? { |role| ROLES.include?(role) }
```
Comment (bcardiff) [inline app/models/collaborator.rb]: why blank roles are valid?

File: app/models/invitation.rb [hunk 2]
```diff
@@ -57,4 +55,10 @@ def creator_is_at_least_collaborator
       end
     end
   end
+
+  def roles_are_valid
+    unless roles.blank? || roles.all? { |role| Collaborator::ROLES.include?(role) }
```
Comment (bcardiff) [inline app/models/invitation.rb]: ditto, blank roles are valid.
